src/backend/distributed/commands/multi_copy.c

```diff
@@ -902,32 +904,25 @@ OpenCopyConnections(CopyStmt *copyStatement, ShardConnections *shardConnections,
 
 		if (PQresultStatus(result) != PGRES_COPY_IN)
 		{
-			ReportConnectionError(connection, WARNING);
-			MarkRemoteTransactionFailed(connection, true);
-
-			PQclear(result);
-
-			/* failed placements will be invalidated by transaction machinery */
-			failedPlacementList = lappend(failedPlacementList, placement);
-			continue;
+			ReportResultError(connection, result, ERROR);
 		}
 
 		PQclear(result);
 		connectionList = lappend(connectionList, connection);
 	}
 
 	/* if all placements failed, error out */
-	if (list_length(failedPlacementList) == list_length(finalizedPlacementList))
+	if (failedPlacementCount == list_length(finalizedPlacementList))
 	{
-		ereport(ERROR, (errmsg("could not find any active placements")));
+		ereport(ERROR, (errmsg("could not connect to any active placements")));
 	}
 
 	/*
 	 * If stopOnFailure is true, we just error out and code execution should
 	 * never reach to this point. This is the case for reference tables and
 	 * copy from worker nodes.
 	 */
-	Assert(!stopOnFailure || list_length(failedPlacementList) == 0);
+	Assert(!stopOnFailure || failedPlacementCount == 0);
```

Comment:
this is sort of obsolete now. if stopOnFailure is set failedPlacementCount is never incremented